CardParts/src/Classes/Card Parts/CardPartOrientedView.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  CardPartOrientedView.swift
+//  CardParts
+//
+//  Created by Tumer, Deniz on 6/13/18.
+//
+
+import Foundation
+
+public enum Orientation {
+    case top
+    case bottom
+}
+
+/**
+ * This card part allows for oriented elements. It takes a list of card part views that can be oriented towards the top or bottom of a view.
+ * Although a stack view has the power to orient elements this oriented card part goes beyond the alignment and distribution types that a stack view can offer.
+ */
+public class CardPartOrientedView: UIView, CardPartView {
+    
+    public var margins: UIEdgeInsets = CardParts.theme.cardPartMargins
+    
+    private var orientation: Orientation
+    private var cardParts: [CardPartView]
+    
+    public init(cardParts: [CardPartView], orientation: Orientation) {
+        self.orientation = orientation
+        self.cardParts = cardParts
+        
+        super.init(frame: CGRect.zero)
+        
+        // add spacer to subviews
+        let spacer = UIView()
+        spacer.translatesAutoresizingMaskIntoConstraints = false
+        spacer.setContentHuggingPriority(.defaultHigh, for: .vertical)
+        self.addSubview(spacer)
+        
+        var prevCardPart: CardPartView = self
+        
+        // go through each card part and link the top margins to the bottom of the previous card part
+        for (ndx, cardPart) in cardParts.enumerated() {
+            let cardPart = cardParts[ndx]
```

Comment:
You don't need to fetch `cardParts[ndx]` that's why you have `for (ndx, cardPart)`.. that will give you the index and the element.

Reply:
fixed this in most recent commit